benchmarks/benchmarks.c

```diff
@@ -55,3 +54,188 @@ void benchmarkFillup ()
 		}
 	}
 }
+
+#ifdef ELEKTRA_ENABLE_OPTIMIZATIONS
+
+// needed for the opmphmRandom ()
+#include "../src/libs/elektra/opmphm.c"
```

Comment:
Yes, it would be used only in tests. Maybe you can at least separate the random function in a separate file?

Btw. should we really use randr_r? It is only in POSIX 2001 and declared obsolete in POSIX 2008. Maybe we use directly the algorithm of the paper you have found? (drand48 is not thread-safe).